packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -494,77 +494,25 @@ end
             continue;
           }
           final String binaryName = globals.fs.path.basenameWithoutExtension(podFrameworkName);
-          if (boolArg('universal')) {
-            globals.fsUtils.copyDirectorySync(
-              podProduct as Directory,
-              modeDirectory.childDirectory(podFrameworkName),
-            );
-            final List<String> lipoCommand = <String>[
-              ...globals.xcode.xcrunCommand(),
-              'lipo',
-              '-create',
-              globals.fs.path.join(podProduct.path, binaryName),
-              if (mode == BuildMode.debug)
-                simulatorBuildConfiguration
-                  .childDirectory(binaryName)
-                  .childDirectory(podFrameworkName)
-                  .childFile(binaryName)
-                  .path,
-              '-output',
-              modeDirectory.childDirectory(podFrameworkName).childFile(binaryName).path
-            ];
-
-            final RunResult pluginsLipoResult = await globals.processUtils.run(
-              lipoCommand,
-              workingDirectory: outputDirectory.path,
-              allowReentrantFlutter: false,
-            );
-
-            if (pluginsLipoResult.exitCode != 0) {
-              throwToolExit(
-                'Unable to create universal $binaryName.framework: ${buildPluginsResult.stderr}',
-              );
-            }
-          }
 
-          if (boolArg('xcframework')) {
-            final List<String> xcframeworkCommand = <String>[
-              ...globals.xcode.xcrunCommand(),
-              'xcodebuild',
-              '-create-xcframework',
-              '-framework',
-              podProduct.path,
-              if (mode == BuildMode.debug)
-                '-framework',
-              if (mode == BuildMode.debug)
-                simulatorBuildConfiguration
-                  .childDirectory(binaryName)
+          final List<Directory> frameworks = <Directory>[
+            podProduct as Directory,
+            if (mode == BuildMode.debug)
+              simulatorBuildConfiguration
+                  .childDirectory(builtProduct.basename)
```

Comment:
@1amageek This is the line that is similar to your suggestion in https://github.com/flutter/flutter/pull/66541, though I used an existing local instead of making a new one.